### ISSUE TITLE
Add VPN Payload (com.apple.vpn.managed)

### ISSUE THREAD

**Issue by apizz** [erikberglund/ProfileManifests#32]
Per https://developer.apple.com/business/documentation/Configuration-Profile-Reference.pdf

**Comment by apizz** [comment]
This payload is crazy complicated.  The manifest for this contained within the Apple Configurator 2 has a ton going on and at the end of the day, does do a better job at visually presenting only the applicable preferences given a selected VPN type. See screenshots below:

![screen shot 2019-02-23 at 11 50 16 am](https://user-images.githubusercontent.com/11789931/53289200-1f6b6c80-3761-11e9-826c-154f0366a714.png)
**vs.**
![screen shot 2019-02-23 at 12 02 30 pm](https://user-images.githubusercontent.com/11789931/53289348-f946cc00-3762-11e9-87d4-53736f00a09c.png)

Notably, there are several `pfm_exclude` arrays that reference nested preferences within a larger preference dictionary.  This is because there are certain preference keys that have the same name but are contained within different dictionaries.  For example, a VPN type of IKEv2 (iOS only) has a `pfm_exclude` that references the `DisconnectOnIdle` preference within the dictionary.  It does this by separating the dictionary and subkey with a period.  The applicable portion of the manifest dictionary is below, notice the `pfm_target`:

```
		<dict>
			<key>pfm_description</key>
			<string>Length of time to wait before disconnecting an on demand connection</string>
			<key>pfm_exclude</key>
			<array>
				<dict>
					<key>pfm_target_conditions</key>
					<array>
						<dict>
							<key>pfm_n_range_list</key>
							<array>
								<integer>1</integer>
							</array>
							<key>pfm_target</key>
							<string>IKEv2.DisconnectOnIdle</string>
						</dict>
					</array>
				</dict>
			</array>
			<key>pfm_name</key>
			<string>DisconnectOnIdleTimer</string>
			<key>pfm_title</key>
			<string>Disconnect on Idle time</string>
			<key>pfm_type</key>
			<string>integer</string>
		</dict>
	</array>
	<key>pfm_title</key>
	<string>IKEv2</string>
	<key>pfm_type</key>
	<string>dictionary</string>
</dict>
```

Interesting as well is the use of dictionaries for the `pfm_default` key, which allows you to define subkey preference defaults at the dictionary key level, and all in one place, rather than at the contained individual preference key level.  Example below:
```
<key>pfm_default</key>
<dict>
	<key>DiffieHellmanGroup</key>
	<integer>14</integer>
	<key>EncryptionAlgorithm</key>
	<string>AES-256</string>
	<key>IntegrityAlgorithm</key>
	<string>SHA2-256</string>
	<key>LifeTimeInMinutes</key>
	<integer>1440</integer>
</dict>
```

Here is the link to my fork with the manifest has it currently exists: https://github.com/apizz/ProfileManifests/commit/32bd3a8fac42316912da94fd5943ce24a1ee1da5

**Comment by erikberglund** [comment]
Yes, this is why I've waited with this payload. And it will have to wait a bit more as I need to add the new method of editing nested arrays first.

**Comment by relgit** [comment]
This issue has been the longest open on the project, which makes me very happy to announce that alongside our update earlier today of iMazing Profile Editor, which includes a major overhaul to the VPN section, making it clearer, more organized, and more workable, we are now committing the manifest that powers it to the project.

@apizz I just pushed branch [`digidna_vpn`](https://github.com/ProfileCreator/ProfileManifests/tree/digidna_vpn) where we can test and make any necessary changes to the manifest as we discussed last June, before eventually merging it to `master`.

**Comment by kevinmcox** [comment]
@relgit @apizz can this related issue be closed as well?

**Comment by kevinmcox** [comment]
@relgit the profile loads fine in PC. However there is something weird going on with the `Role` and `WgQuickConfig` keys.

In PC they show up at the top all the time and can't be interacted with.

In Imazing they also seem to act weird to me. They aren't displayed by default, but if I search for them they appear in the UI.

![Image](https://github.com/user-attachments/assets/5a190f59-e788-45a8-b915-1e2f92fa2cec)
